lib/browser/api/dialog.js

```diff
@@ -70,195 +55,135 @@ const checkAppInitialized = function () {
   }
 }
 
-module.exports = {
-  showOpenDialog: function (...args) {
-    checkAppInitialized()
-
-    let [window, options, callback] = parseArgs(...args)
-
-    if (options == null) {
-      options = {
-        title: 'Open',
-        properties: ['openFile']
-      }
-    }
-
-    let {buttonLabel, defaultPath, filters, properties, title, message, securityScopedBookmarks = false} = options
-
-    if (properties == null) {
-      properties = ['openFile']
-    } else if (!Array.isArray(properties)) {
-      throw new TypeError('Properties must be an array')
-    }
+const setupDialog = (type, {
+    window,
+    buttonLabel = '',
+    defaultPath = '',
+    filters = [],
+    properties = ['openFile'],
+    title = '',
+    message = '',
+    securityScopedBookmarks = false,
+    callback,
+    nameFieldLabel = '',
+    showsTagField = true
+  }) => {
+  if (!Array.isArray(properties)) {
+    throw new TypeError('Properties must be an array')
+  }
 
-    let dialogProperties = 0
+  let dialogProperties = 0
+  // set up dialogProperties for showOpenDialog
+  if (type === Type.OPEN) {
     for (const prop in fileDialogProperties) {
       if (properties.includes(prop)) {
         dialogProperties |= fileDialogProperties[prop]
       }
     }
+  }
 
-    if (title == null) {
-      title = ''
-    } else if (typeof title !== 'string') {
-      throw new TypeError('Title must be a string')
-    }
+  // check parameter types
+  if (typeof title !== 'string') {
+    throw new TypeError('Title must be a string')
+  }
+  if (typeof buttonLabel !== 'string') {
+    throw new TypeError('Button label must be a string')
+  }
+  if (typeof defaultPath !== 'string') {
+    throw new TypeError('Default path must be a string')
+  }
+  if (typeof message !== 'string') {
+    throw new TypeError('Message must be a string')
+  }
 
-    if (buttonLabel == null) {
-      buttonLabel = ''
-    } else if (typeof buttonLabel !== 'string') {
-      throw new TypeError('Button label must be a string')
+  // set up extra parameters for showSaveDialog
```

Comment:
This comment might be clearer as "check extra parameters for showSaveDialog".  When I read it as setting up the parameters I wondered why this block wasn't up with the other test on type,
```js
if (type === Type.OPEN) {
```

However when I saw that the code was checking parameter types, it made more sense in its placement